### PR TITLE
drop primitive serializer support for data:: types

### PR DESCRIPTION
**Pull request by akva2**
Downstream of https://github.com/OPM/opm-common/pull/3142

### CODE REVIEW

File: opm/simulators/utils/ParallelRestart.cpp [hunk 1]
```diff
@@ -31,35 +31,11 @@
 #include <dune/common/parallel/mpitraits.hh>
 #if HAVE_MPI
 #include <ebos/eclmpiserializer.hh>
-#endif
```
Comment (bska) [inline opm/simulators/utils/ParallelRestart.cpp:34]: I'm probably blind, but I didn't see this `#endif` reinstated somewhere else.  Is that intentional?

Comment (akva2) [reply opm/simulators/utils/ParallelRestart.cpp:34]: no, rebase meh. fixed.

**Review by bska — APPROVED** [review]
Thanks.  The one downside I see to removing the separate unit tests is that it is a lot harder to single-step into a macro in a debugger if the test fails.  On the other hand I suppose we can always insert a dedicated test case locally if that situation occurs.

I'll merge into master.

**Comment by atgeirr** [comment]
Having written part of the code deleted here just weeks ago, I support this removal!

**Comment by blattms** [comment]
Good for you. Not always easy, though. :+1:

**Pull request merged**